### ISSUE TITLE
Declare variable and immediately ending block with variable

### ISSUE THREAD

**Issue by Havvy** [Manishearth/rust-clippy#104]
```rust
fn foo (): V {
  let v: V = bar();
  v
}
```

Should produce a warning stating that the `let` is unnecessary and to just return `bar()` directly from the block.

Should probably apply to all blocks that can return values (so `if`, `fn`, and `match` blocks plus any I'm not remembering).

**Comment by llogiq** [comment]
So we'd match blocks with at least 2 expressions, where the next-to-last statement is a let-binding and the last is an ExprPath with the same path as the let-binding.

**Comment by Manishearth** [comment]
Note that semicolon-less exprs are stored in a different field of the block (they're not statements)

**Comment by llogiq** [comment]
True, so we'd match the last statement against the result expression, if any.
